diff --git a/CMakeLists.txt b/CMakeLists.txt
index c7791812..92934f39 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -20,6 +20,7 @@ set(PRISMATIC_ENABLE_CLI 1 CACHE BOOL PRISMATIC_ENABLE_GPU)
 set(PRISMATIC_ENABLE_DOUBLE_PRECISION 0 CACHE BOOL PRISMATIC_ENABLE_DOUBLE_PRECISION)
 set(PRISMATIC_ENABLE_PYPRISMATIC 0 CACHE BOOL PRISMATIC_ENABLE_PYPRISMATIC)
 set(PRISMATIC_USE_HDF5_STATIC 0 CACHE BOOL PRISMATIC_USE_HDF5_STATIC)
+set(OUTPUT_NAME prismatic CACHE STRING OUTPUT_NAME)
 
 #set (CMAKE_BUILD_TYPE DEBUG)
 if (PRISMATIC_ENABLE_GPU)
@@ -168,6 +169,7 @@ if (PRISMATIC_ENABLE_CLI)
 #        		   ${Boost_LIBRARY_DIRS}
                    ${FFTW_LIBRARIES}
                    ${HDF5_LIBRARIES})
+    set_target_properties(prismatic PROPERTIES OUTPUT_NAME ${OUTPUT_NAME})
 endif (PRISMATIC_ENABLE_CLI)
 
 if(APPLE)
@@ -272,22 +274,17 @@ if(PRISMATIC_ENABLE_PYPRISMATIC)
     target_link_libraries(pyprismatic_core
                     ${CMAKE_THREAD_LIBS_INIT}
                     ${FFTW_LIBRARIES}
-                    ${HDF5_LIBRARIES}
-                    ${PYTHON_LIBRARIES})
-
-    set_target_properties(
-    pyprismatic_core
-    PROPERTIES
-        PREFIX ""
-        OUTPUT_NAME "core"
-        LINKER_LANGUAGE CXX
-    )
-
-    if(NOT UNIX)
-    set_target_properties(
-        pyprismatic_core
-        PROPERTIES
-            SUFFIX ".pyd"
-    )
-    endif(NOT UNIX)
-endif(PRISMATIC_ENABLE_PYPRISMATIC)
\ No newline at end of file
+                    ${HDF5_LIBRARIES})
+
+    set_target_properties(pyprismatic_core PROPERTIES PREFIX "" OUTPUT_NAME "core" LINKER_LANGUAGE CXX)
+
+    if (WIN32)
+        target_link_libraries(pyprismatic_core ${PYTHON_LIBRARIES})
+        set_target_properties(pyprismatic_core PROPERTIES SUFFIX ".pyd")
+    elseif (APPLE)
+        set_target_properties(pyprismatic_core PROPERTIES LINK_FLAGS "-undefined dynamic_lookup")
+        set_target_properties(pyprismatic_core PROPERTIES SUFFIX ".so")
+    endif ()
+    # else Linux: dont link
+
+endif(PRISMATIC_ENABLE_PYPRISMATIC)

