diff --git a/CMakeLists.txt b/CMakeLists.txt
index c7791812..45577973 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -73,8 +73,8 @@ find_package (Boost REQUIRED)
 find_package (FFTW3 REQUIRED)
 
 if(PRISMATIC_ENABLE_PYPRISMATIC)
-    find_package (PythonInterp 3.5 REQUIRED)
-    find_package (PythonLibs 3.5 REQUIRED)
+    find_package (PythonInterp REQUIRED)
+    find_package (PythonLibs REQUIRED)
 endif(PRISMATIC_ENABLE_PYPRISMATIC)
 
 if (PRISMATIC_USE_HDF5_STATIC)
diff --git a/setup.py b/setup.py
index ab80de27..20a181a7 100644
--- a/setup.py
+++ b/setup.py
@@ -119,11 +119,7 @@ class cmake_build_ext(build_ext):
                     '-DCMAKE_RUNTIME_OUTPUT_DIRECTORY_{}={}'.format(cfg.upper(), extdir),
                 ]
                 # Assuming that Visual Studio and MinGW are supported compilers
-                if self.compiler.compiler_type == 'msvc':
-                    cmake_args += [
-                        '-DCMAKE_GENERATOR_PLATFORM=%s' % plat,
-                    ]
-                else:
+                if self.compiler.compiler_type != 'msvc':
                     cmake_args += [
                         '-G', 'MinGW Makefiles',
                     ]
-- 